seq2science/rules/merge_replicates.smk

```diff
@@ -60,6 +60,8 @@ if "replicate" in samples:
             ],
             **config,
         )
+        if len(input_files["reps"]) == 0:
+            raise ValueError("Something went wrong, and we tried to merge replicates but there were no replicates?!")
```

Comment:
Nice for debugging, but not very nice to leave in. In `renamefastq_PE` I solved this by adding an impossible input if the input was empty. This input can double as a debug message, if snakemake calls an `missinginputException` on it.

Reply:
I think this is personal. I prefer to raise an error immediately than have snakemake try to make some file from some "impossible" output. It really should not happen! It's not like people put in the wrong sample name, this is a real problem if it happens.

Also using "impossible" output might result in weird bugs/side effects later, especially in combination with generic rules like unzip.

Reply:
an impossible input prevents the rule from being used. no weird side effects. Something like
```
if len(input_files["reps"]) == 0:
    return "Something went wrong, and we tried to merge replicates but there were no replicates?!"
```

The reason I used this solution is because (I think) snakemake runs all these rules while trying to generate the DAG. So it might "try" the rule, then run into the error, even though it wouldn't use the rule in the end. Not sure if we test for this scenario.